2-mandatory/4-stocks.js

```diff
@@ -34,61 +34,80 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
         Functions can help with this!
 */
 function getAveragePrices(closingPricesForAllStocks) {
-    // TODO
+  let totalAvg = [];
+  for (let eachStock of closingPricesForAllStocks) {
+    let avg = 0;
+    let total = 0;
+    for (let price of eachStock) {
+      total += price;
+      avg = total / eachStock.length;
+      avg = Math.round(avg * 100) / 100;
+    }
+    totalAvg.push(avg);
+  }
+
+  return totalAvg;
 }
 
-/*
-    We also want to see what the change in price is from the first day to the last day for each stock.
-    Implement the below function, which
-        - Takes this CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, it's an array of arrays)
-        - Returns an array containing the price change over the last 5 days for each stock.
-            For example, the first element of the resulting array should contain Apple’s (aapl) price change for the last 5 days.
-            In this example it would be:
-                (Apple's price on the 5th day) - (Apple's price on the 1st day) = 172.99 - 179.19 = -6.2
-    The price change value should be rounded to 2 decimal places, and should be a number (not a string)
-*/
 function getPriceChanges(closingPricesForAllStocks) {
-    // TODO
+  let changeInPrice = [];
+  let changeForEachStock = 0;
+  let rounded;
+  for (let prices of closingPricesForAllStocks) {
+    changeForEachStock = prices[prices.length - 1] - prices[0];
+    rounded = Math.round(changeForEachStock * 100) / 100;
+    changeInPrice.push(rounded);
+  }
+  return changeInPrice;
 }
-
-/*
-    As part of a financial report, we want to see what the highest price was for each stock in the last 5 days.
-    Implement the below function, which
-        - Takes 2 parameters:
-            - the CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, it's an array of arrays)
-            - the STOCKS array
-        - Returns an array of strings describing what the highest price was for each stock.
-            For example, the first element of the array should be: "The highest price of AAPL in the last 5 days was 180.33"
-            The test will check for this exact string.
-    The stock ticker should be capitalised.
-    The price should be shown with exactly 2 decimal places.
-*/
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
+  // TODO
+  let newStocks = [];
+  let sorted = [];
+  let message = "";
+  for (let stock of closingPricesForAllStocks) {
+    stock.sort((a, b) => b - a);
+    sorted.push(stock);
+  }
+  for (let i = 0; i < stocks.length; i++) {
+    let roundedHighest = sorted[i][0].toFixed(2);
```

Comment:
Refer to my comment above. You may try to replace **sorted** with **closingPricesForAllStocks**